### DISCUSSION TITLE
Ability to easily toggle between a set of themes (light/dark)

### DISCUSSION THREAD

**Discussion by EriKWDev** [helix-editor/helix#3488] [Ideas]
#### Describe your feature request
I usually use helix on my laptop and when it is bright outside it is nice to use a light theme. At home, in the dark, it is much nicer with a darker theme.

It would be nice if there was some simple binding to toggle between themes that I have defined.

```toml
# config.toml
themes = [ "monokai_pro_spectrum", "catppuccin_latte" ] # ...or more
```
With proposed bindings:
- Next theme \<Space\> \<t\> \<n\>
- Previous themes: \<Space\> \<t\> \<p\>
- Select theme #n:  \<Space\> \<t\> \<0/1/2/3/4/5/6/7/8/9\>

It would be nice if this decision was saved between sessions

---

We could keep compatability by making the old `theme = "..."` still work unless the new `themes` is present. I would not mind other bindings if you come up with better ones, but I think the proposed ones seem to align with others.

I also prefer this over something like having two `dark-theme = "..."`, `light-theme = "..."` since this approach is more general.

**Comment by the-mikedavis** [comment]
This might fall under https://github.com/helix-editor/helix/issues/2849 but it seems slightly different since it's about configuration directly rather than enabling/disabling features.

Persisting between sessions I think is covered by https://github.com/helix-editor/helix/issues/401.

I think it would be really elegant if this could be accomplished with custom keymaps. That would need a more general purpose configuration language (TOML is temporary).

As a workaround, you can hack at least the `<space>t<0/1/2/3...>` part with

```toml
# ~/.config/helix/config.toml
[keys.normal.space.t]
"1" = ":theme monokai_pro_spectrum"
"2" = ":theme catppuccin_latte"
# ...
```

You can also remove all themes from the `~/.config/helix/runtime/themes` except the ones you would like to toggle between and then `:theme <tab><tab>...`. That would remove most of the friction from the current setup.

Overall though this seems like a somewhat specific use-case and I'm not sure it warrants building a feature around it. I may be out of touch on this but I don't imagine many people switch themes very often.

**Reply by EriKWDev** [reply]
Ye I don't know how niche this is either, but those keybindings will help a bit.

**Comment by kirawi** [comment]
Seems like a duplicate of https://github.com/helix-editor/helix/issues/2158

**Reply by the-mikedavis** [reply]
I think it's subtley different: this one is more about manual toggling. But it's possible that the use-case would be served equally well by #2158
